packages/core/src/hydration/annotate.ts

```diff
@@ -197,6 +207,10 @@ export function annotateForHydration(appRef: ApplicationRef, doc: Document) {
   const serializedViews = serializedViewCollection.getAll();
   const transferState = injector.get(TransferState);
   transferState.set(NGH_DATA_KEY, serializedViews);
+  if (shouldReplayEvents) {
+    return eventTypesToReplay;
+  }
+  return undefined;
```

Comment:
In some cases we might be returning an empty `Set`. We can add a check here and only return a `Set` if it's not empty. Also see a comment below (where we invoke the `annotateForHydration` function).
```suggestion
  return eventTypesToReplay.size > 0  ? eventTypesToReplay : undefined;
```

Reply:
done